modules/rostests/winetests/CMakeLists.txt

```diff
@@ -1,8 +1,16 @@
 
 add_definitions(-D__ROS_LONG64__)
 
-if (MSVC AND ARCH STREQUAL "amd64")
-    add_compile_flags("/wd4312")
+if (MSVC)
+    add_compile_flags("/wd4090") # C4090: 'function': different 'const' qualifiers
+    add_compile_flags("/wd4146") # C4146: unary minus operator applied to unsigned type, result still unsigned
+    add_compile_flags("/wd4189") # C4189: 'x': local variable is initialized but not referenced
+    add_compile_flags("/wd4267") # C4267: '=': conversion from 'size_t' to 'int', possible loss of data
+    add_compile_flags("/wd4305") # C4305: '=': truncation from 'double' to 'FLOAT'
+    if (ARCH STREQUAL "amd64")
+        add_compile_flags("/wd4101") # C4101: 'x': unreferenced local variable
+        add_compile_flags("/wd4312") # C4312: 'type cast': conversion from 'unsigned int' to 'char *' of greater size
+    endif()
```

Comment:
:+1: It would be amusing to send the list of compilation warnings to the wine guys ^^

Reply:
Feel free to do so 😄